trunk/SUAVE/Components/Energy/Converters/Propeller.py

```diff
@@ -279,11 +294,324 @@ def spin(self,conditions):
+        Cp    = np.zeros_like(Ct)
+        power = np.zeros_like(Ct)        
+        for i in range(len(Vv)):
+            if -1. <Vv[i][0] <1.: # vertical/axial flight
+                Cp[i]       = (kappa*(Ct[i]**1.5)/(2**.5))+sigma*Cd0/8.
+                power[i]    = Cp[i]*(rho[i]*(n[i]*n[i]*n[i])*(D*D*D*D*D))
+                torque[i]   = power[i]/omega[i]  
+            else:  
+                power[i]    = torque[i]*omega[i]   
+                Cp[i]       = power[i]/(rho[i]*(n[i]*n[i]*n[i])*(D*D*D*D*D)) 
+  
+        thrust[conditions.propulsion.throttle[:,0] <=0.0] = 0.0
+        power[conditions.propulsion.throttle[:,0]  <=0.0] = 0.0
+        thrust[omega<0.0] = - thrust[omega<0.0]
+
+        etap     = V*thrust/power     
+        
+        conditions.propulsion.etap = etap
+        
+        # store data
+        results_conditions = Data     
+        outputs   = results_conditions(
+            n_blades                  = B,
+            R                         = R,
+            D                         = D,
+            number_sections           = N,
+            radius_distribution       = np.linspace(Rh ,R, N),
+            chord_distribution        = c,     
+            twist_distribution        = beta,            
+            r0                        = r,
+            thrust_angle              = theta,
+            speed_of_sound            = conditions.freestream.speed_of_sound,
+            density                   = conditions.freestream.density,
+            velocity                  = Vv, 
+            vt                        = vt, 
+            va                        = va, 
+            drag_coefficient          = Cd,
+            lift_coefficient          = Cl,       
+            omega                     = omega,          
+            
+            blade_dT_dR               = rho*(Gamma*(Wt-epsilon*Wa)),   
+            blade_dT_dr               = rho*(Gamma*(Wt-epsilon*Wa))*R,  
+            blade_T_distribution      = rho*(Gamma*(Wt-epsilon*Wa))*deltar, 
+            blade_T                   = thrust/B,  
+            Ct                        = 0, 
+            Cts                       = 0, 
+            
+            blade_dQ_dR               = rho*(Gamma*(Wa+epsilon*Wt)*r), 
+            blade_dQ_dr               = rho*(Gamma*(Wa+epsilon*Wt)*r)*R,
+            blade_Q_distribution      = rho*(Gamma*(Wa+epsilon*Wt)*r)*deltar,
+            blade_Q                   = torque/B,   
+            Cq                        = 0, 
+            
+            power                     = power,
+            
+            mid_chord_aligment        = self.mid_chord_aligment     
+        ) 
+        
+        return thrust, torque, power, Cp, outputs  , etap  
+    
+
+    def spin_variable_pitch(self,conditions):
+        """Analyzes a propeller given geometry and operating conditions.
+
+        Assumptions:
+        per source
+
+        Source:
+        Qprop theory document
```

Comment:
This still needs a more specific source.

Reply:
Source updated for Propeller.py and Rotor.py